### PR TITLE
Added a check on the TRH version in HDF5LIBS_TestDumpRecord

### PR DESCRIPTION
**Pull request by bieryAtFnal**
This will be helpful in avoiding confusion if users happen to run the new version of the utlitiy on older data files.

To test these changes, run the HDF5LIBS_TestDumpRecord application on raw data files produced with fddaq-v4.3.0 or earlier and confirm that an error message is printed out.

### CODE REVIEW

File: test/apps/HDF5LIBS_TestDumpRecord.cpp [hunk 1]
```diff
@@ -99,6 +99,18 @@ main(int argc, char** argv)
...
+      if (trh_ptr->get_header().version != TriggerRecordHeaderData::s_trigger_record_header_version) {
+        std::cout << std::endl;
+        std::cout << "ERROR: The specified data file was written with a version of the DUNE-DAQ software that "
+                  << "used a different version" << std::endl << "       of the TriggerRecordHeader "
+                  << "than this application (built with the current version of the software) is expecting."
+                  << std::endl;
+        std::cout << "Please use a version of the software that is compatible with the data file." << std::endl;
+        std::cout << "(Expected TRH version "
+                  << trh_ptr->get_header().version << " and found version "
+                  << TriggerRecordHeaderData::s_trigger_record_header_version << ".)" << std::endl;
```
Comment (aeoranday) [inline test/apps/HDF5LIBS_TestDumpRecord.cpp]: This language is flipped: "Expected" should be the one attached to `TriggerRecordHeaderData` and "found" should be from the `get_header()`.

**Review by aeoranday — CHANGES_REQUESTED** [review]
There is a language mistake in the expected vs found version printout. I thought I was compiling incorrectly for a moment!

**Comment by bieryAtFnal** [comment]
wow, sorry, thanks for catching that.
Fixed now.

**Review by aeoranday — APPROVED** [review]
LGTM.

**Pull request merged**